lib/src/widgets/input/input_types/input_date_picker/date_picker.dart

```diff
@@ -0,0 +1,111 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_form/utils/translation_service.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:intl/intl.dart';
+import '../../../../../flutter_form.dart';
+
+/// Generates a [TextFormField] for passwords. It requires a [FlutterFormInputController]
+/// as the [controller] parameter and an optional [Widget] as [label]
+class DateTimeInputField extends ConsumerStatefulWidget {
+  const DateTimeInputField(
+      {Key? key,
+      required this.inputType,
+      required this.controller,
+      this.label,
+      this.showIcon = true,
+      this.icon = Icons.calendar_today,
+      required this.dateFormat})
+      : super(
+          key: key,
+        );
+  final FlutterFormDateTimeType inputType;
+  final FlutterFormInputController controller;
+  final DateFormat dateFormat;
+  final bool showIcon;
+  final IconData icon;
+  final Widget? label;
+  @override
+  ConsumerState<DateTimeInputField> createState() => _DateInputFieldState();
+}
+
+class _DateInputFieldState extends ConsumerState<DateTimeInputField> {
+  @override
+  Widget build(BuildContext context) {
+    String Function(String, {List<String>? params}) _ =
+        getTranslator(context, ref);
+
+    Future<String> getInputFromUser(FlutterFormDateTimeType inputType) async {
+      String userInput = '';
+      switch (inputType) {
+        case FlutterFormDateTimeType.date:
+          DateTime? unformatted = await showDatePicker(
+            context: context,
+            initialDate: DateTime.now(),
+            firstDate: DateTime.now(),
```

Comment:
Make the firstDate and lastdate configurable with the standard of 1000 days previous and 1000 days past datetime.now